src/SectorView.cpp

```diff
@@ -507,6 +731,65 @@ void SectorView::Draw3D()
 		}
 		DrawRouteLines(modelview);
 	}
+}
+
+// function is used to pass label parameters from lua
+void SectorView::SetLabelParams(std::string fontName, int fontSize, float gap, Color highlight, Color shade)
+{
+	m_labels.fontName = fontName;
+	m_labels.fontSize = fontSize;
+	m_labels.gap = gap;
+	m_labels.colors.highlight = IM_COL32(highlight.r, highlight.g, highlight.b, highlight.a);
+	m_labels.colors.shade = IM_COL32(shade.r, shade.g, shade.b, shade.a);
+}
+
+// function runs from lua, inside imgui context
+void SectorView::DrawLabels()
+{
+	// sort the labels to draw them starting from the farthest
+	std::sort(m_labels.array.begin(), m_labels.array.end(), [](const std::unique_ptr<Label> &a, const std::unique_ptr<SectorView::Label> &b) {
+		return a->Depth() > b->Depth();
+	});
+
+	ImVec2 mpos = ImGui::GetMousePos();
+	const size_t NOT_FOUND = m_labels.array.size();
+	// the index of the label the cursor is hovering over
+	size_t hovered_i = NOT_FOUND;
+
+	m_drawList->_ResetForNewFrame();
+	StarLabel::InitFonts(*this, *m_drawList);
+	FactionLabel::InitFonts(*this, *m_drawList);
+	m_drawList->PushClipRectFullScreen();
+
+	// iterate over the labels starting with the closest one until we find the one over which the cursor is hanging
+	if (!ImGui::IsWindowHovered(ImGuiHoveredFlags_AnyWindow)) {
+		for (int i = m_labels.array.size() - 1; i >= 0; --i) {
+			Label &label = *m_labels.array[i];
+			if (hovered_i == NOT_FOUND && !m_rotateWithMouseButton && label.Hovered(mpos.x, mpos.y)) {
+				hovered_i = i;
+				if (ImGui::IsMouseReleased(0)) label.OnClick();
+				break;
+			}
+		}
+	}
+
+	// draw labels
+	for (size_t i = 0; i < m_labels.array.size(); ++i) {
+		Label &label = *m_labels.array[i];
+		if (i != hovered_i) {
+			m_drawList->AddDrawCmd();
+			m_drawList->CmdBuffer.back().PrimDepth = -label.Depth() * 1.01;
+			label.Draw(*m_drawList);
+		}
+	}
+
+	// draw the label over which the cursor is hovering the most recent
+	if (hovered_i != NOT_FOUND) {
+		Label &hovered = *m_labels.array[hovered_i];
+		m_drawList->AddDrawCmd();
+		m_drawList->CmdBuffer.back().PrimDepth = 1.0f;
```

Comment:
If I correctly remember how this is implemented, a PrimDepth of 1.0 is the very back of the scene and all UI elements are drawn with a depth of 0. I could be misremembering this, I'd have to dig back and check the commit.

Reply:
I have empirically established that with such depth, the object seems to be drawn on top of all.